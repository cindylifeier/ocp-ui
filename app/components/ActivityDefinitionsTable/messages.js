/*
 * ActivityDefinitionsTable Messages
 *
 * This contains all the text for the ActivityDefinitionsTable component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  tableHeaderColumnDisplayName: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnDisplayName',
    defaultMessage: 'Display Name',
  },
  tableHeaderColumnTopic: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnTopic',
    defaultMessage: 'Topic',
  },
  tableHeaderColumnEffectiveStart: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnEffectiveStart',
    defaultMessage: 'Effective Start',
  },
  tableHeaderColumnEffectiveEnd: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnEffectiveEnd',
    defaultMessage: 'Effective End',
  },
  tableHeaderColumnDescription: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnDescription',
    defaultMessage: 'Description',
  },
  tableHeaderColumnStatus: {
    id: 'ocpui.components.ActivityDefinitionsTable.tableHeaderColumnStatus',
    defaultMessage: 'Status',
  },
  edit: {
    id: 'ocpui.components.ActivityDefinitionsTable.edit',
    defaultMessage: 'Edit',
  },
  noResultsFound: {
    id: 'ocpui.components.ActivityDefinitionsTable.noResultsFound',
    defaultMessage: 'No activity definitions found.',
  },
});
