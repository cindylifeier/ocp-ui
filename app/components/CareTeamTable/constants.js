
/**
 *
 * @type {string}
 */
export const EXPANDED_TABLE_COLUMNS = '1fr 1fr 1fr 2fr 1fr 1fr 1fr .5fr';
export const SUMMARIZED_TABLE_COLUMNS = 'repeat(3, 1fr) .5fr';
