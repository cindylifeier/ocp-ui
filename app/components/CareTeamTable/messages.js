/*
 * CareTeamTable Messages
 *
 * This contains all the text for the CareTeamTable component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  columnHeaderName: {
    id: 'ocpui.components.CareTeamTable.columnHeaderName',
    defaultMessage: 'Name',
  },
  columnHeaderStatus: {
    id: 'ocpui.components.CareTeamTable.columnHeaderStatus',
    defaultMessage: 'Status',
  },
  columnHeaderAction: {
    id: 'ocpui.components.CareTeamTable.columnHeaderAction',
    defaultMessage: 'Action',
  },
  columnHeaderCategories: {
    id: 'ocpui.components.CareTeamTable.columnHeaderCategories',
    defaultMessage: 'Categories',
  },
  columnHeaderParticipantsAndRoles: {
    id: 'ocpui.components.CareTeamTable.columnHeaderCategories',
    defaultMessage: 'Participant / Role',
  },
  columnHeaderStartDate: {
    id: 'ocpui.components.CareTeamTable.columnHeaderStartDate',
    defaultMessage: 'Start Date',
  },
  columnHeaderEndDate: {
    id: 'ocpui.components.CareTeamTable.columnHeaderEndDate',
    defaultMessage: 'End Date',
  },
  columnHeaderReason: {
    id: 'ocpui.components.CareTeamTable.columnHeaderReason',
    defaultMessage: 'Reason',
  },
  menuItemEdit: {
    id: 'ocpui.components.CareTeamTable.menuItemEdit',
    defaultMessage: 'Edit',
  },
  menuItemRemove: {
    id: 'ocpui.components.CareTeamTable.menuItemRemove',
    defaultMessage: 'Remove',
  },
});
