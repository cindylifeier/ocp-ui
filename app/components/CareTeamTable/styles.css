.menuItem {
  display: inline-block;
}

.iconButton svg {
  fill: rgb(51, 51, 51) !important;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
}

.iconButton:hover svg {
  fill: #26A69A !important;
}

.iconButtonGridContainer {
  display: grid;
  grid-template-columns: 1fr 50px;
}

.iconButtonGridItem {
  grid-column-start: 2;
  grid-column-end: 3;
  grid-row-start: 1;
  grid-row-end: 2;
}
