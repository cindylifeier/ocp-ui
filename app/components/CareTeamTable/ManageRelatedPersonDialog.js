import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { DialogActions, DialogContent, DialogTitle } from 'material-ui-next/Dialog';
import StyledDialog from 'components/StyledDialog';
import StyledFlatButton from 'components/StyledFlatButton';
import messages from './messages';

function ManageRelatedPersonDialog(props) {
  const { dialogOpen, onDialogClose } = props;
  return (
    <div>
      <StyledDialog open={dialogOpen} fullWidth>
        <DialogTitle>
          <FormattedMessage {...messages.manageRelatedPersonDialogTitle} />
        </DialogTitle>
        <DialogContent>
          <h1>Test</h1>
        </DialogContent>
        <DialogActions>
          <StyledFlatButton onClick={onDialogClose}>
            <FormattedMessage {...messages.cancelButton} />
          </StyledFlatButton>
        </DialogActions>
      </StyledDialog>
    </div>
  );
}

ManageRelatedPersonDialog.propTypes = {
  dialogOpen: PropTypes.bool.isRequired,
  onDialogClose: PropTypes.func.isRequired,
};

export default ManageRelatedPersonDialog;
