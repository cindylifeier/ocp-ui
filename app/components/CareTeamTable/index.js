/**
 *
 * CareTeamTable
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import isEmpty from 'lodash/isEmpty';
import sizeMeHOC from 'utils/SizeMeUtils';
import Table from 'components/Table';
import TableHeader from 'components/TableHeader';
import TableHeaderColumn from 'components/TableHeaderColumn';
import TableRow from 'components/TableRow';
import {
  EXPANDED_TABLE_COLUMNS,
  SUMMARIZED_TABLE_COLUMNS,
} from 'components/CareTeamTable/constants';
import TableRowColumn from 'components/TableRowColumn';
import NavigationIconMenu from 'components/NavigationIconMenu';
import messages from './messages';

function CareTeamTable({ elements, relativeTop, manageCareTeamUrl, isExpanded }) {
  function createTableHeaders() {
    const columns = isExpanded ? EXPANDED_TABLE_COLUMNS : SUMMARIZED_TABLE_COLUMNS;
    return (
      <TableHeader columns={columns} relativeTop={relativeTop}>
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderName} /></TableHeaderColumn>
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderStatus} /></TableHeaderColumn>
        {isExpanded &&
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderCategories} /></TableHeaderColumn>
        }
        {isExpanded &&
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderParticipantsAndRoles} /></TableHeaderColumn>
        }
        { isExpanded &&
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderStartDate} /></TableHeaderColumn>
        }
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderEndDate} /></TableHeaderColumn>
        {isExpanded &&
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderReason} /></TableHeaderColumn>
        }
        <TableHeaderColumn><FormattedMessage {...messages.columnHeaderAction} /></TableHeaderColumn>
      </TableHeader>
    );
  }

  function createTableRows(id, name, statusDisplay, categoryDisplay, participants, startDate, endDate, reasonDisplay, menuItems) {
    const columns = isExpanded ? EXPANDED_TABLE_COLUMNS : SUMMARIZED_TABLE_COLUMNS;
    return (
      <TableRow key={id} columns={columns}>
        <TableRowColumn>{name}</TableRowColumn>
        <TableRowColumn>{statusDisplay}</TableRowColumn>
        {isExpanded &&
        <TableRowColumn>{categoryDisplay}</TableRowColumn>
        }
        {isExpanded &&
        <TableRowColumn>
          {!isEmpty(participants) && participants
            .map(({ memberId, memberFirstName, memberLastName, memberName, roleDisplay }) => (
              <div key={memberId}>
                {`${[memberFirstName, memberLastName, memberName].filter((value) => !isEmpty(value)).join(' ')}${isEmpty(roleDisplay) ? '' : ` / ${roleDisplay}`}`}
              </div>))
          }
        </TableRowColumn>
        }
        { isExpanded &&
        <TableRowColumn>{startDate}</TableRowColumn>
        }
        <TableRowColumn>{endDate}</TableRowColumn>
        {isExpanded &&
        <TableRowColumn>{reasonDisplay}</TableRowColumn>
        }
        <TableRowColumn>
          <NavigationIconMenu menuItems={menuItems} />
        </TableRowColumn>
      </TableRow>
    );
  }
  return (
    <div>
      <Table>
        {createTableHeaders()}
        {!isEmpty(elements) && elements.map(({ id, name, statusDisplay, categoryDisplay, participants, subjectId, startDate, endDate, reasonDisplay }) => {
          const menuItems = [{
            primaryText: <FormattedMessage {...messages.menuItemEdit} />,
            linkTo: {
              pathname: `${manageCareTeamUrl}/${id}`,
              search: `?patientId=${subjectId}`,
            },
          }, {
            primaryText: <FormattedMessage {...messages.menuItemRemove} />,
            disabled: true,
          }];
          return createTableRows(id, name, statusDisplay, categoryDisplay, participants, startDate, endDate, reasonDisplay, menuItems);
        })}
      </Table>
    </div>
  );
}

CareTeamTable.propTypes = {
  isExpanded: PropTypes.bool,
  relativeTop: PropTypes.number.isRequired,
  manageCareTeamUrl: PropTypes.string.isRequired,
  elements: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    reasonCode: PropTypes.string,
    reasonDisplay: PropTypes.string,
    statusCode: PropTypes.string,
    statusDisplay: PropTypes.string,
    categoryCode: PropTypes.string,
    categoryDisplay: PropTypes.string,
    subjectId: PropTypes.string.isRequired,
    subjectFirstName: PropTypes.string.isRequired,
    subjectLastName: PropTypes.string.isRequired,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    participants: PropTypes.arrayOf(PropTypes.shape({
      roleCode: PropTypes.string,
      roleDisplay: PropTypes.string,
      memberId: PropTypes.string.isRequired,
      memberFirstName: PropTypes.string,
      memberLastName: PropTypes.string,
      memberName: PropTypes.string,
      memberType: PropTypes.string.isRequired,
      onBehalfOfId: PropTypes.string,
      onBehalfOfName: PropTypes.string,
    })),
  })),
};

export default sizeMeHOC(CareTeamTable);
