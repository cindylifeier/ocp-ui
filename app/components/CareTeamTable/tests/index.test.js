// import React from 'react';
// import { shallow } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import { FormattedMessage } from 'react-intl';

import CareTeamTable from '../index';
import messages from '../messages';
import TableHeaderColumn from '../../TableHeaderColumn';

const mockElements = [{
  id: '111111',
  name: 'My Care Team One',
  statusCode: 'active',
  statusDisplay: 'Active',
  reasonCodes: [{
    code: '109006',
    display: 'Anxiety disorder of childhood OR adolescence',
  }, {
    code: '122003',
    display: 'Choroidal hemorrhage',
  },
  ],
  categoryCode: 'encounter',
  categoryDisplay: 'Encounter',
  subjectId: '11111',
  subjectFirstName: 'Adam',
  subjectLastName: 'Smith',
  startDate: '01/01/2017',
  endDate: '02/01/2017',
  participants: [{
    roleCode: '112247003',
    roleDisplay: 'Medical doctor',
    memberId: '700551',
    memberFirstName: 'Gregory', // if not organization
    memberLastName: 'House', // if not organization
    memberType: 'practitioner', // can be one of `patient`, `practitioner`, `organization` or `relatedPerson`
    onBehalfOfId: '143191',
    onBehalfOfName: 'Great Cross Hospital',

  }, {
    roleCode: '394765007',
    roleDisplay: 'Prison practice',
    memberId: '650858',
    memberName: 'Great Cross Hospital', // if organization
    memberType: 'organization', // can be one of `patient`, `practitioner`, `organization` or `relatedPerson`
    onBehalfOfId: '143191',
    onBehalfOfName: 'Great Cross Hospital',
  },
  ],
}, {
  id: '222222',
  name: 'My Care Team Two',
  statusCode: 'suspended',
  statusDisplay: 'Suspended',
  reasonCodes: [{
    code: '134006',
    display: 'Decreased hair growth',
  },
  ],
  categoryCode: 'condition',
  categoryDisplay: 'Condition',
  subjectId: '11111',
  subjectFirstName: 'Adam',
  subjectLastName: 'Smith',
  startDate: '04/01/2017',
  endDate: '08/01/2017',
  participants: [{
    roleCode: '113157001',
    roleDisplay: 'Grand-mother',
    memberId: '121212',
    memberFirstName: 'Granny', // if not organization
    memberLastName: 'Smith', // if not organization
    memberType: 'relatedPerson', // can be one of `patient`, `practitioner`, `organization` or `relatedPerson`
  },
  ],
},
];

describe('<CareTeamTable />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should contain table header columns', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains(
      <TableHeaderColumn><FormattedMessage {...messages.columnHeaderId} /></TableHeaderColumn>)).toBe(true);
    expect(renderedComponent.contains(
      <TableHeaderColumn><FormattedMessage {...messages.columnHeaderName} /></TableHeaderColumn>)).toBe(true);
    expect(renderedComponent.contains(
      <TableHeaderColumn><FormattedMessage {...messages.columnHeaderStatus} /></TableHeaderColumn>)).toBe(true);
    expect(renderedComponent.contains(
      <TableHeaderColumn><FormattedMessage {...messages.columnHeaderCategories} /></TableHeaderColumn>)).toBe(true);
    expect(renderedComponent.contains(
      <TableHeaderColumn><FormattedMessage {...messages.columnHeaderParticipantsAndRoles} /></TableHeaderColumn>)).toBe(true);
  });

  it('should contain participants', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains('Gregory House / Medical doctor')).toBe(true);
    expect(renderedComponent.contains('Great Cross Hospital / Prison practice')).toBe(true);
    expect(renderedComponent.contains('Granny Smith / Grand-mother')).toBe(true);
  });

  it('should contain care team names', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains('My Care Team One')).toBe(true);
    expect(renderedComponent.contains('My Care Team Two')).toBe(true);
  });

  it('should contain categories', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains('Encounter')).toBe(true);
    expect(renderedComponent.contains('Condition')).toBe(true);
  });

  it('should contain statuses', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains('Active')).toBe(true);
    expect(renderedComponent.contains('Suspended')).toBe(true);
  });

  it('should contain ids', () => {
    // Act
    const renderedComponent = shallow(<CareTeamTable elements={mockElements} />);

    // Assert
    expect(renderedComponent.contains('111111')).toBe(true);
    expect(renderedComponent.contains('222222')).toBe(true);
  });
});
