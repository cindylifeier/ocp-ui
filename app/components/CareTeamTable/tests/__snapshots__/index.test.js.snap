// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CareTeamTable /> should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Table>
        <TableHeader>
            <TableHeaderColumn>
                <FormattedMessage
                    defaultMessage="ID"
                    id="ocpui.components.CareTeamTable.columnHeaderId"
                    values={Object {}}
                />
            </TableHeaderColumn>
            <TableHeaderColumn>
                <FormattedMessage
                    defaultMessage="Name"
                    id="ocpui.components.CareTeamTable.columnHeaderName"
                    values={Object {}}
                />
            </TableHeaderColumn>
            <TableHeaderColumn>
                <FormattedMessage
                    defaultMessage="Status"
                    id="ocpui.components.CareTeamTable.columnHeaderStatus"
                    values={Object {}}
                />
            </TableHeaderColumn>
            <TableHeaderColumn>
                <FormattedMessage
                    defaultMessage="Categories"
                    id="ocpui.components.CareTeamTable.columnHeaderCategories"
                    values={Object {}}
                />
            </TableHeaderColumn>
            <TableHeaderColumn>
                <FormattedMessage
                    defaultMessage="Participant / Role"
                    id="ocpui.components.CareTeamTable.columnHeaderCategories"
                    values={Object {}}
                />
            </TableHeaderColumn>
        </TableHeader>
        <TableRow>
            <TableRowColumn>
                111111
            </TableRowColumn>
            <TableRowColumn>
                My Care Team One
            </TableRowColumn>
            <TableRowColumn>
                Active
            </TableRowColumn>
            <TableRowColumn>
                Encounter
            </TableRowColumn>
            <TableRowColumn>
                <ul>
                    <li>
                        Gregory House / Medical doctor
                    </li>
                    <li>
                        Great Cross Hospital / Prison practice
                    </li>
                </ul>
            </TableRowColumn>
        </TableRow>
        <TableRow>
            <TableRowColumn>
                222222
            </TableRowColumn>
            <TableRowColumn>
                My Care Team Two
            </TableRowColumn>
            <TableRowColumn>
                Suspended
            </TableRowColumn>
            <TableRowColumn>
                Condition
            </TableRowColumn>
            <TableRowColumn>
                <ul>
                    <li>
                        Granny Smith / Grand-mother
                    </li>
                </ul>
            </TableRowColumn>
        </TableRow>
    </Table>
</div>,
  "nodes": Array [
    <div>
      <Table>
            <TableHeader>
                  <TableHeaderColumn>
                        <FormattedMessage
                              defaultMessage="ID"
                              id="ocpui.components.CareTeamTable.columnHeaderId"
                              values={Object {}}
                        />
                  </TableHeaderColumn>
                  <TableHeaderColumn>
                        <FormattedMessage
                              defaultMessage="Name"
                              id="ocpui.components.CareTeamTable.columnHeaderName"
                              values={Object {}}
                        />
                  </TableHeaderColumn>
                  <TableHeaderColumn>
                        <FormattedMessage
                              defaultMessage="Status"
                              id="ocpui.components.CareTeamTable.columnHeaderStatus"
                              values={Object {}}
                        />
                  </TableHeaderColumn>
                  <TableHeaderColumn>
                        <FormattedMessage
                              defaultMessage="Categories"
                              id="ocpui.components.CareTeamTable.columnHeaderCategories"
                              values={Object {}}
                        />
                  </TableHeaderColumn>
                  <TableHeaderColumn>
                        <FormattedMessage
                              defaultMessage="Participant / Role"
                              id="ocpui.components.CareTeamTable.columnHeaderCategories"
                              values={Object {}}
                        />
                  </TableHeaderColumn>
            </TableHeader>
            <TableRow>
                  <TableRowColumn>
                        111111
                  </TableRowColumn>
                  <TableRowColumn>
                        My Care Team One
                  </TableRowColumn>
                  <TableRowColumn>
                        Active
                  </TableRowColumn>
                  <TableRowColumn>
                        Encounter
                  </TableRowColumn>
                  <TableRowColumn>
                        <ul>
                              <li>
                                    Gregory House / Medical doctor
                              </li>
                              <li>
                                    Great Cross Hospital / Prison practice
                              </li>
                        </ul>
                  </TableRowColumn>
            </TableRow>
            <TableRow>
                  <TableRowColumn>
                        222222
                  </TableRowColumn>
                  <TableRowColumn>
                        My Care Team Two
                  </TableRowColumn>
                  <TableRowColumn>
                        Suspended
                  </TableRowColumn>
                  <TableRowColumn>
                        Condition
                  </TableRowColumn>
                  <TableRowColumn>
                        <ul>
                              <li>
                                    Granny Smith / Grand-mother
                              </li>
                        </ul>
                  </TableRowColumn>
            </TableRow>
      </Table>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CareTeamTable
        elements={
                Array [
                        Object {
                          "categoryCode": "encounter",
                          "categoryDisplay": "Encounter",
                          "endDate": "02/01/2017",
                          "id": "111111",
                          "name": "My Care Team One",
                          "participants": Array [
                            Object {
                              "memberFirstName": "Gregory",
                              "memberId": "700551",
                              "memberLastName": "House",
                              "memberType": "practitioner",
                              "onBehalfOfId": "143191",
                              "onBehalfOfName": "Great Cross Hospital",
                              "roleCode": "112247003",
                              "roleDisplay": "Medical doctor",
                            },
                            Object {
                              "memberId": "650858",
                              "memberName": "Great Cross Hospital",
                              "memberType": "organization",
                              "onBehalfOfId": "143191",
                              "onBehalfOfName": "Great Cross Hospital",
                              "roleCode": "394765007",
                              "roleDisplay": "Prison practice",
                            },
                          ],
                          "reasonCodes": Array [
                            Object {
                              "code": "109006",
                              "display": "Anxiety disorder of childhood OR adolescence",
                            },
                            Object {
                              "code": "122003",
                              "display": "Choroidal hemorrhage",
                            },
                          ],
                          "startDate": "01/01/2017",
                          "statusCode": "active",
                          "statusDisplay": "Active",
                          "subjectFirstName": "Adam",
                          "subjectId": "11111",
                          "subjectLastName": "Smith",
                        },
                        Object {
                          "categoryCode": "condition",
                          "categoryDisplay": "Condition",
                          "endDate": "08/01/2017",
                          "id": "222222",
                          "name": "My Care Team Two",
                          "participants": Array [
                            Object {
                              "memberFirstName": "Granny",
                              "memberId": "121212",
                              "memberLastName": "Smith",
                              "memberType": "relatedPerson",
                              "roleCode": "113157001",
                              "roleDisplay": "Grand-mother",
                            },
                          ],
                          "reasonCodes": Array [
                            Object {
                              "code": "134006",
                              "display": "Decreased hair growth",
                            },
                          ],
                          "startDate": "04/01/2017",
                          "statusCode": "suspended",
                          "statusDisplay": "Suspended",
                          "subjectFirstName": "Adam",
                          "subjectId": "11111",
                          "subjectLastName": "Smith",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "elements": Array [
            Object {
              "categoryCode": "encounter",
              "categoryDisplay": "Encounter",
              "endDate": "02/01/2017",
              "id": "111111",
              "name": "My Care Team One",
              "participants": Array [
                Object {
                  "memberFirstName": "Gregory",
                  "memberId": "700551",
                  "memberLastName": "House",
                  "memberType": "practitioner",
                  "onBehalfOfId": "143191",
                  "onBehalfOfName": "Great Cross Hospital",
                  "roleCode": "112247003",
                  "roleDisplay": "Medical doctor",
                },
                Object {
                  "memberId": "650858",
                  "memberName": "Great Cross Hospital",
                  "memberType": "organization",
                  "onBehalfOfId": "143191",
                  "onBehalfOfName": "Great Cross Hospital",
                  "roleCode": "394765007",
                  "roleDisplay": "Prison practice",
                },
              ],
              "reasonCodes": Array [
                Object {
                  "code": "109006",
                  "display": "Anxiety disorder of childhood OR adolescence",
                },
                Object {
                  "code": "122003",
                  "display": "Choroidal hemorrhage",
                },
              ],
              "startDate": "01/01/2017",
              "statusCode": "active",
              "statusDisplay": "Active",
              "subjectFirstName": "Adam",
              "subjectId": "11111",
              "subjectLastName": "Smith",
            },
            Object {
              "categoryCode": "condition",
              "categoryDisplay": "Condition",
              "endDate": "08/01/2017",
              "id": "222222",
              "name": "My Care Team Two",
              "participants": Array [
                Object {
                  "memberFirstName": "Granny",
                  "memberId": "121212",
                  "memberLastName": "Smith",
                  "memberType": "relatedPerson",
                  "roleCode": "113157001",
                  "roleDisplay": "Grand-mother",
                },
              ],
              "reasonCodes": Array [
                Object {
                  "code": "134006",
                  "display": "Decreased hair growth",
                },
              ],
              "startDate": "04/01/2017",
              "statusCode": "suspended",
              "statusDisplay": "Suspended",
              "subjectFirstName": "Adam",
              "subjectId": "11111",
              "subjectLastName": "Smith",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Table>
                    <TableHeader>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="ID"
                                                  id="ocpui.components.CareTeamTable.columnHeaderId"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Name"
                                                  id="ocpui.components.CareTeamTable.columnHeaderName"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Status"
                                                  id="ocpui.components.CareTeamTable.columnHeaderStatus"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Categories"
                                                  id="ocpui.components.CareTeamTable.columnHeaderCategories"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Participant / Role"
                                                  id="ocpui.components.CareTeamTable.columnHeaderCategories"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                    </TableHeader>
                    <TableRow>
                              <TableRowColumn>
                                        111111
                              </TableRowColumn>
                              <TableRowColumn>
                                        My Care Team One
                              </TableRowColumn>
                              <TableRowColumn>
                                        Active
                              </TableRowColumn>
                              <TableRowColumn>
                                        Encounter
                              </TableRowColumn>
                              <TableRowColumn>
                                        <ul>
                                                  <li>
                                                            Gregory House / Medical doctor
                                                  </li>
                                                  <li>
                                                            Great Cross Hospital / Prison practice
                                                  </li>
                                        </ul>
                              </TableRowColumn>
                    </TableRow>
                    <TableRow>
                              <TableRowColumn>
                                        222222
                              </TableRowColumn>
                              <TableRowColumn>
                                        My Care Team Two
                              </TableRowColumn>
                              <TableRowColumn>
                                        Suspended
                              </TableRowColumn>
                              <TableRowColumn>
                                        Condition
                              </TableRowColumn>
                              <TableRowColumn>
                                        <ul>
                                                  <li>
                                                            Granny Smith / Grand-mother
                                                  </li>
                                        </ul>
                              </TableRowColumn>
                    </TableRow>
          </Table>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Table>
                    <TableHeader>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="ID"
                                                  id="ocpui.components.CareTeamTable.columnHeaderId"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Name"
                                                  id="ocpui.components.CareTeamTable.columnHeaderName"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Status"
                                                  id="ocpui.components.CareTeamTable.columnHeaderStatus"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Categories"
                                                  id="ocpui.components.CareTeamTable.columnHeaderCategories"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                              <TableHeaderColumn>
                                        <FormattedMessage
                                                  defaultMessage="Participant / Role"
                                                  id="ocpui.components.CareTeamTable.columnHeaderCategories"
                                                  values={Object {}}
                                        />
                              </TableHeaderColumn>
                    </TableHeader>
                    <TableRow>
                              <TableRowColumn>
                                        111111
                              </TableRowColumn>
                              <TableRowColumn>
                                        My Care Team One
                              </TableRowColumn>
                              <TableRowColumn>
                                        Active
                              </TableRowColumn>
                              <TableRowColumn>
                                        Encounter
                              </TableRowColumn>
                              <TableRowColumn>
                                        <ul>
                                                  <li>
                                                            Gregory House / Medical doctor
                                                  </li>
                                                  <li>
                                                            Great Cross Hospital / Prison practice
                                                  </li>
                                        </ul>
                              </TableRowColumn>
                    </TableRow>
                    <TableRow>
                              <TableRowColumn>
                                        222222
                              </TableRowColumn>
                              <TableRowColumn>
                                        My Care Team Two
                              </TableRowColumn>
                              <TableRowColumn>
                                        Suspended
                              </TableRowColumn>
                              <TableRowColumn>
                                        Condition
                              </TableRowColumn>
                              <TableRowColumn>
                                        <ul>
                                                  <li>
                                                            Granny Smith / Grand-mother
                                                  </li>
                                        </ul>
                              </TableRowColumn>
                    </TableRow>
          </Table>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CareTeamTable
    elements={
        Array [
            Object {
              "categoryCode": "encounter",
              "categoryDisplay": "Encounter",
              "endDate": "02/01/2017",
              "id": "111111",
              "name": "My Care Team One",
              "participants": Array [
                Object {
                  "memberFirstName": "Gregory",
                  "memberId": "700551",
                  "memberLastName": "House",
                  "memberType": "practitioner",
                  "onBehalfOfId": "143191",
                  "onBehalfOfName": "Great Cross Hospital",
                  "roleCode": "112247003",
                  "roleDisplay": "Medical doctor",
                },
                Object {
                  "memberId": "650858",
                  "memberName": "Great Cross Hospital",
                  "memberType": "organization",
                  "onBehalfOfId": "143191",
                  "onBehalfOfName": "Great Cross Hospital",
                  "roleCode": "394765007",
                  "roleDisplay": "Prison practice",
                },
              ],
              "reasonCodes": Array [
                Object {
                  "code": "109006",
                  "display": "Anxiety disorder of childhood OR adolescence",
                },
                Object {
                  "code": "122003",
                  "display": "Choroidal hemorrhage",
                },
              ],
              "startDate": "01/01/2017",
              "statusCode": "active",
              "statusDisplay": "Active",
              "subjectFirstName": "Adam",
              "subjectId": "11111",
              "subjectLastName": "Smith",
            },
            Object {
              "categoryCode": "condition",
              "categoryDisplay": "Condition",
              "endDate": "08/01/2017",
              "id": "222222",
              "name": "My Care Team Two",
              "participants": Array [
                Object {
                  "memberFirstName": "Granny",
                  "memberId": "121212",
                  "memberLastName": "Smith",
                  "memberType": "relatedPerson",
                  "roleCode": "113157001",
                  "roleDisplay": "Grand-mother",
                },
              ],
              "reasonCodes": Array [
                Object {
                  "code": "134006",
                  "display": "Decreased hair growth",
                },
              ],
              "startDate": "04/01/2017",
              "statusCode": "suspended",
              "statusDisplay": "Suspended",
              "subjectFirstName": "Adam",
              "subjectId": "11111",
              "subjectLastName": "Smith",
            },
          ]
    }
/>,
}
`;
